desktop/main.go

```diff
@@ -182,137 +86,208 @@ func start(a *astilectron.Astilectron, w []*astilectron.Window, _ *astilectron.M
 		}
 	}()
 
-	// start the gateway
-	gateway.Host.Start(fmt.Sprintf("127.0.0.1:%s", core.GetRandomPort()))
-
-	// save off the server address
-	gatewayAddr = fmt.Sprintf("http://%s", gateway.Host.Addr())
-
-	// sleep for a bit on the landing screen, it feels better
-	time.Sleep(SleepOnLoad)
-
-	// send cookie info to front-end
-	sendData("login", map[string]interface{}{
-		"name":    "SessionId",
-		"value":   "not used",
-		"gateway": gatewayAddr,
-	})
+	// setup and start the apis
+	gateway.Host = &gateway.Gateway{
+		Node: node,
+	}
+	node.StartApi(node.Config().Addresses.API, true)
+	gateway.Host.Start(node.Config().Addresses.Gateway)
 
-	// check if we're configured yet
-	threads := node.Threads()
-	if len(threads) > 0 {
-		// load threads for UI
-		var threadsJSON []map[string]interface{}
-		for _, thrd := range threads {
-			threadsJSON = append(threadsJSON, map[string]interface{}{
-				"id":   thrd.Id,
-				"name": thrd.Name,
-			})
-		}
+	// wait for node to come online
+	<-node.OnlineCh()
 
-		// reveal
-		sendPreReady()
-		window.Hide()
-		expandWindow()
-		sendData("ready", map[string]interface{}{
-			"threads": threadsJSON,
-		})
-		window.Show()
-		window.Focus()
+	return nil
+}
 
-	} else {
-		// get qr code for setup
-		qr, pk, err := getQRCode()
-		if err != nil {
-			astilog.Error(err)
-			return err
+func stopNode() error {
+	if err := node.Stop(); err != nil {
+		astilog.Error(err)
+		if err == core.ErrStopped {
+			return nil
 		}
-		sendData("setup", map[string]interface{}{
-			"qr": qr,
-			"pk": pk,
-		})
+		return err
+	}
+	if err := node.StopApi(); err != nil {
+		return err
+	}
+	if err := gateway.Host.Stop(); err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func sendMessage(name string) {
-	window.SendMessage(map[string]string{"name": name})
-}
+func start(app *astilectron.Astilectron, _ []*astilectron.Window, _ *astilectron.Menu, t *astilectron.Tray, m *astilectron.Menu) error {
+	// remove the dock icon
+	var d = app.Dock()
+	d.Hide()
 
-func sendData(name string, data map[string]interface{}) {
-	data["name"] = name
-	window.SendMessage(data)
-}
+	// get homedir
+	home, err := homedir.Dir()
+	if err != nil {
+		astilog.Fatal(fmt.Errorf("get homedir failed: %s", err))
+	}
+
+	// ensure app support folder is created
+	var appDir string
+	if runtime.GOOS == "darwin" {
+		appDir = filepath.Join(home, "Library", "Application Support", "Textile")
+	} else {
+		appDir = filepath.Join(home, ".textile")
+	}
+	if err := os.MkdirAll(appDir, 0755); err != nil {
+		astilog.Fatal(fmt.Errorf("create app dir failed: %s", err))
+	}
+	repoPath := filepath.Join(appDir, "repo")
+
+	// run init if needed
+	if !fsrepo.IsInitialized(repoPath) {
+		accnt := keypair.Random()
+		initc := core.InitConfig{
+			Account:   accnt,
+			RepoPath:  repoPath,
+			LogToDisk: true,
+			GatewayAddr: fmt.Sprintf("127.0.0.1:5052"),
```

Comment:
Had these as random ports at one point, switched back to fixed so its easier to have 'defaults' for clients. Worth a chat though to make sure this is a good idea @sanderpick .

Reply:
sounds good to me